purchase_sale_stock_inter_company/models/purchase_order.py

```diff
@@ -22,11 +22,18 @@ def _prepare_sale_order_data(
         )
         if delivery_address:
             new_order.update({"partner_shipping_id": delivery_address.id})
+
+        potential_warehouse = self.env["stock.warehouse"].search(
+            [("partner_id", "=", self.partner_id.id)]
+        )
```

Comment:
To prevent singleton error:
```suggestion
        limit=1)
```